src/main/java/command/ClearCommand.java

```diff
@@ -13,12 +15,35 @@ public ClearCommand(Printer printer, CheatSheetList cheatSheetList, DataFileDest
         super(printer);
         this.cheatSheetList = cheatSheetList;
         this.fileDestroyer = fileDestroyer;
+        if (printer.getDisplayingHelpMessages()) {
+            printer.printClearCommandHelpMessage();
+        }
     }
 
     @Override
     public void execute() {
-        fileDestroyer.executeFunction();
-        printer.printClearCheatSheetMessage(cheatSheetList.getSize());
-        cheatSheetList.clear();
+        if (isClearCommandConfirmed()) {
+            fileDestroyer.executeFunction();
+            printer.printClearCheatSheetMessage(cheatSheetList.getSize());
+            cheatSheetList.clear();
+        } else {
+            System.out.println("CLEAR command aborted.");
+        }
+    }
+
+    private boolean isClearCommandConfirmed() {
+        while (true) {
+            System.out.print("\nDo you wish to continue (y/n): ");
```

Comment:
I think just to keep it consistent with the delete confirmation, they should be able to type both `y` and `yes`, case-insensitive.

Reply:
Ahh right, I didn't realise aldo did the delete part until I finish the clear part. I'll change it later